document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/PostgresQueryParser.java

```diff
@@ -1,104 +1,112 @@
 package org.hypertrace.core.documentstore.postgres.query.v1;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.Optional;
-import org.hypertrace.core.documentstore.expression.type.FilterTypeExpression;
+import lombok.Getter;
 import org.hypertrace.core.documentstore.postgres.Params;
 import org.hypertrace.core.documentstore.postgres.Params.Builder;
 import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresFilterTypeExpressionVisitor;
 import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresGroupTypeExpressionVisitor;
 import org.hypertrace.core.documentstore.postgres.query.v1.vistors.PostgresSelectTypeExpressionVisitor;
 import org.hypertrace.core.documentstore.query.Pagination;
 import org.hypertrace.core.documentstore.query.Query;
-import org.hypertrace.core.documentstore.query.SelectionSpec;
 import org.hypertrace.core.documentstore.query.SortingSpec;
 
 public class PostgresQueryParser {
   private static String NOT_YET_SUPPORTED = "Not yet supported %s";
-  private Builder paramsBuilder;
   private final String collection;
 
-  public PostgresQueryParser(String collection) {
-    this.collection = collection;
+  @Getter private Builder paramsBuilder;
+
+  @Getter private Query query;
+
+  @Getter private Map<String, String> pgSelections; // alias to field name
+
+  private void resetParser(Query query) {
+    this.query = query;
+    paramsBuilder = Params.newBuilder();
+    pgSelections = new HashMap<>();
   }
 
-  public Builder getParamsBuilder() {
-    return paramsBuilder;
+  public PostgresQueryParser(String collection) {
+    this.collection = collection;
   }
 
   public String parse(Query query) {
     // prepare selection and form clause
     // TODO : add impl for selection + form clause for unwind
-    StringBuilder sqlBuilder = new StringBuilder(String.format("SELECT * FROM %s", collection));
-    paramsBuilder = Params.newBuilder();
+    StringBuilder sqlBuilder = new StringBuilder();
+    this.resetParser(query);
 
-    // selection clause
-    Optional<String> selectionClause = parseSelection(query.getSelections());
-    if (selectionClause.isPresent()) {
-      sqlBuilder = new StringBuilder();
-      sqlBuilder.append(String.format("SELECT %s FROM %s", selectionClause.get(), collection));
-    }
     // where clause
-    Optional<String> whereFilter = parseFilter(query.getFilter());
+    Optional<String> whereFilter = parseFilter();
```

Comment:
Note: we need alias only in aggregation filter, so where clause is parsed first, and alias is prepared later in selection clause for aggregation filter.